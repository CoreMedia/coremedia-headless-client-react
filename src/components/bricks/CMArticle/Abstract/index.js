// @flow
import { P } from '../../../basic/Paragraph';
import { media, getColors } from '../../../../styles/themes/utils';

const Abstract = P.extend`
  font-weight: 300;
  padding: 2em;
  color: ${props => getColors(props.theme.colors, props.color).fg};
  background-color: ${props => getColors(props.theme.colors, props.color).bgsolid};
  font-size: 1.2em;
  ${props => media(props.theme.breakpoints.tablet)`
    font-size: 1.3em;
  `};
  ${props => media(props.theme.breakpoints.desktop)`
    font-size: 1.4em;
  `};
  ${props => media(props.theme.breakpoints.large)`
    font-size: 1.6em;
  `};
`;

Abstract.displayName = 'Abstract';

Abstract.defaultProps = {
  theme: {
    name: 'default',
    breakpoints: {
      tablet: 768,
      desktop: 992,
      large: 1280,
    },
    colors: {
      default: {
        fg: '#fafafa',
        bgsolid: 'rgba(239, 223, 15, 1)',
        bgtrans: 'rgba(239, 223, 15, 0.7)',
        huerotation: '5deg',
      },
    },
  },
};

export default Abstract;
