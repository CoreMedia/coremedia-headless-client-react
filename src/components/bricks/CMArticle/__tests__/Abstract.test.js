// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Abstract from '../Abstract';

describe('Abstract Component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Abstract>Content</Abstract>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
