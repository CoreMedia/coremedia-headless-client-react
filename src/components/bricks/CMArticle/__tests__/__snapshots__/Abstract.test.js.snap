// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abstract Component should render correctly 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: 'Simplon Norm Light','Lucida Sans','Lucida Sans Unicode','Lucida Grande',Arial, Helvetica,sans-serif;
  font-weight: 400;
  line-height: 1.4;
  font-size: 1em;
  font-weight: 300;
  padding: 2em;
  color: #fafafa;
  background-color: rgba(239,223,15,1);
  font-size: 1.2em;
}

@media (min-width:48em) {
  .c0 {
    font-size: 1em;
  }
}

@media (min-width:62em) {
  .c0 {
    font-size: 1.1em;
  }
}

@media (min-width:80em) {
  .c0 {
    font-size: 1.2em;
  }
}

@media (min-width:48em) {
  .c0 {
    font-size: 1.3em;
  }
}

@media (min-width:62em) {
  .c0 {
    font-size: 1.4em;
  }
}

@media (min-width:80em) {
  .c0 {
    font-size: 1.6em;
  }
}

<p
  className="c0"
>
  Content
</p>
`;
